bots/s3_only_allow_ssl.py

```diff
@@ -0,0 +1,178 @@
+"""
+## s3_only_allow_ssl
+What it does: Ensure that S3 Buckets enforce encryption of data transfers using Secure Sockets Layer (SSL)
+Usage: AUTO: s3_only_allow_ssl
+Note: The bot looks at the bucket policy and adds to the current policy the missing actions(s3:GetObject and s3:PutObject)
+      and the SSL statement.
+      if no policy in the bucket an SSL policy will add to the bucket
+Limitations: none
+"""
+
+import json
+from botocore.exceptions import ClientError
+
+
+def run_action(boto_session, rule, entity, params):
+    # create a s3 sessions
+    s3_client = boto_session.client('s3')
+
+    bucket_name = entity['name']
+    account_number = entity["accountNumber"]
+    policy_bucket = entity['policy']
+
+    try:
+        if policy_bucket == "null":  # s3 does not have a bucket policy
+            policy = json.dumps(make_new_policy(bucket_name, account_number))
+
+        else:  # bucket has a policy
+            res = get_correct_policy(policy_bucket)
+            first_Effect, second_Effect = get_actions(bucket_name, account_number, res)
+            if second_Effect is None:  # if there is only one Effect to add to the bucket policy
+                policy_bucket['Statement'].append(first_Effect)
+                policy = json.dumps(policy_bucket)
+            else:
+                policy_bucket['Statement'].append(first_Effect)
+                policy_bucket['Statement'].append(second_Effect)
+                policy = json.dumps(policy_bucket)
+        print(policy)
+        result = s3_client.put_bucket_policy(
+            Bucket=bucket_name,
+            Policy=policy
+        )
+
+        responseCode = result['ResponseMetadata']['HTTPStatusCode']
```

Comment:
Again, you can avoid using response code

Reply:
removed